### DISCUSSION TITLE
fluentd benchmarks

### DISCUSSION THREAD

**Discussion by eregon** [fluent/fluentd#3893] [Q&A] [answered]
Hello there,
Is there something like standard fluentd benchmarks?

I've found https://github.com/fluent/fluentd-benchmark but those seem to always include networking and multiple processes (which makes it harder to measure/profile/debug) and I'm not sure which one(s) are representative of typical fluentd workloads.

A benchmark for fluentd parsing performance without networking would be quite nice to simplify and have a single process, is there such a benchmark?

**Comment by fujimotos** [comment, marked as answer]
> A benchmark for fluentd parsing performance without networking would be quite nice to simplify and have a single process, is there such a benchmark?

IMO, the easiest way to archive this goal is to use [fluent-plugin-flowcounter-simple](https://github.com/sonots/fluent-plugin-flowcounter-simple).

First, you need to install this plugin using `gem` command:

```console
$ gem install fluentd --no-doc
$ fluent-gem install fluent-plugin-flowcounter-simple 
```

Then, setup the basic configuration as follows:

```console
$ cat fluent.conf
<source>
  @type tail
  path access.log
  format apache2
  tag test.log
  read_from_head true
</source>
<match test.**> 
  @type flowcounter_simple
  unit second
</match>
$ yes '127.0.0.1 - - [08/Sep/2022:15:27:31 +0900] "GET / HTTP/1.1" 200 0 "index.html" "test"' | head -n 10000000 > access.log
```

Now you can run `fluentd`. As show below, you can check how many lines
Fluentd could processed per second in the log:
 
```console
$ fluentd -c fluentd.conf
...
2022-09-13 12:37:12 +0000 [info]: #0 plugin:out_flowcounter_simple	count:45056	indicator:num	unit:second
2022-09-13 12:37:13 +0000 [info]: #0 plugin:out_flowcounter_simple	count:98494	indicator:num	unit:second
2022-09-13 12:37:14 +0000 [info]: #0 plugin:out_flowcounter_simple	count:108973	indicator:num	unit:second
...
```

You can tweak `fluent.conf` as you see fit (e.g. adding some `filter` plugins to
make the pipeline more complex), but the gist is the same here. Enjoy!
